### PR TITLE
add node id to pageserver

### PR DESCRIPTION
**Pull request by LizardWizzard**
This adds node id parameter to pageserver configuration. Also I use a
simple builder to construct pageserver config struct to avoid setting
node id to some temporary invalid value. Some of the changes in test
fixtures are needed to split init and start operations for envrionment.

### CODE REVIEW

**Comment by arssher** [comment]
Let's also add ` -c "id='10'"` to `docker-entrypoint.sh` for console development/testing.

File: pageserver/src/bin/pageserver.rs [hunk 1]
```diff
@@ -115,7 +115,14 @@ fn main() -> Result<()> {
                     option_line
                 )
             })?;
+
             for (key, item) in doc.iter() {
+                if key == "id" {
+                    anyhow::ensure!(
```
Comment (SomeoneToIgnore) [inline pageserver/src/bin/pageserver.rs:121]: NIT: Should we rather WARN log it and skip it instead of failing?

File: pageserver/src/config.rs [hunk 2]
```diff
@@ -106,6 +110,168 @@ pub struct PageServerConf {
...
+    max_file_descriptors: Option<usize>,
+
+    workdir: Option<PathBuf>,
+
+    pg_distrib_dir: Option<PathBuf>,
+
+    auth_type: Option<AuthType>,
+
+    auth_validation_public_key_path: Option<Option<PathBuf>>,
+    remote_storage_config: Option<Option<RemoteStorageConfig>>,
```
Comment (SomeoneToIgnore) [inline pageserver/src/config.rs:136]: > `Option<Option<`

Out of curiosity, what does this semantic mean? 
Would we be able to achieve it with a single `Option<>` wrapper?

File: pageserver/src/config.rs [hunk 3]
```diff
@@ -106,6 +110,168 @@ pub struct PageServerConf {
...
+            gc_horizon: Some(DEFAULT_GC_HORIZON),
+            gc_period: Some(
+                humantime::parse_duration(DEFAULT_GC_PERIOD)
+                    .expect("cannot parse default gc period"),
+            ),
+            superuser: Some(DEFAULT_SUPERUSER.to_string()),
+            page_cache_size: Some(DEFAULT_PAGE_CACHE_SIZE),
+            max_file_descriptors: Some(DEFAULT_MAX_FILE_DESCRIPTORS),
+            workdir: Default::default(),
+            pg_distrib_dir: Some(PathBuf::new()),
```
Comment (SomeoneToIgnore) [inline pageserver/src/config.rs:161]: > `Some(PathBuf::new())`

That is a bit odd to me, is `PathBuf::new()` ever a valid path for the `pg_distrib_dir`?

File: pageserver/src/config.rs [hunk 4]
```diff
@@ -106,6 +110,168 @@ pub struct PageServerConf {
...
+            auth_type: Some(AuthType::Trust),
+            auth_validation_public_key_path: Some(None),
+            remote_storage_config: Some(None),
+            id: Default::default(),
```
Comment (SomeoneToIgnore) [inline pageserver/src/config.rs:165]: > `Default::default()`

NIT: that is not that readable + adds more work for the compiler to infer this, `ZNodeId::default()` is way more appealing to me personally.

**Review by SomeoneToIgnore — APPROVED** [review]

File: pageserver/src/config.rs (hunk 2 above)
Comment (LizardWizzard) [reply pageserver/src/config.rs:136]: Outer option is needed for builder to distinguish between set/not set, the inner one is the desired value in target structure we are building

File: pageserver/src/bin/pageserver.rs (hunk 1 above)
Comment (LizardWizzard) [reply pageserver/src/bin/pageserver.rs:121]: In my opinion node id sort of related to correctness so I would rather fail loudly. It is an error to have one id in config and a pageserver actually running with different id. If you need to forcibly change the value you can edit the config manually.

File: pageserver/src/config.rs (hunk 3 above)
Comment (LizardWizzard) [reply pageserver/src/config.rs:161]: Hmm, I've blindly copied defaults, in the config checking code I see this: https://github.com/zenithdb/zenith/blob/74a0942a773359754e704a662e7d109cd07489a9/pageserver/src/config.rs#L289
so I think I can directly use it as a default without intermediate value

**Review by arssher — APPROVED** [review]

**Pull request merged**